src/scripts.js

```diff
@@ -2,107 +2,209 @@ import './css/styles.css';
 import './images/turing-logo.png';
 import Travelers from './traveler';
 import Trips from './trips';
-import Destinations from './destinations'
-import {travelersData, destinationsData, tripsData} from './apiCalls';
+import Destinations from './destinations';
+import {travelersData, destinationsData, tripsData, addNewTrip, fetchData, tripsApi} from './apiCalls';
 import dayjs from 'dayjs';
 
 //Query Selectors
 const travelerName = document.querySelector('h2')
 const tripsDisplay = document.querySelector('.trips')
 const costThisYear = document.querySelector('#total-cost-this-year')
 const destinationsPicker = document.querySelector('.destinations-picker')
-const datePicker = document.querySelector('.date')
 const durationPicker = document.querySelector('.duration-picker')
 const guestPicker = document.querySelector('.guests-picker')
+const datePicker = document.querySelector('.date-picker')
+const button = document.querySelector('.submit-button')
+const messageBanner = document.querySelector('h3')
+const doubleClickMsg = document.querySelector('h4')
+const form = document.querySelector('.selections')
 
+//Event Listeners
+destinationsPicker.addEventListener('input', checkInput)
+datePicker.addEventListener('input', () => {checkDate(), checkInput()})
+durationPicker.addEventListener('input', checkInput)
+guestPicker.addEventListener('input', checkInput)
 
-const displayedUsersID = Math.floor(Math.random() * 50)
+button.addEventListener('click', checkNumberOfClick)
+window.addEventListener('load', buildFormHelper)
 
-window.addEventListener('load', buildForm)
 
-function buildForm() {
-  preventPickingPastDates()
-  buildDropDown()
-}
+const displayedUsersID = Math.floor(Math.random() * 50)
+let allDestinations;
+let allTrips;
+let allTravelers;
+let newPendingTrip;
+let today;
+
 
 Promise.all([travelersData, destinationsData, tripsData])
   .then((data) => {
     travelerHelper(data[0].travelers);
-    tripsHelper(data[2].trips, data[1].destinations);
+    destinationsHelper(data[1].destinations);
+    tripsHelper(data[2].trips);
   })
-  .catch((error) => console.log(error));
+  .catch((error) => {console.log(error), displayErrorMessage()});
+
+
+//HELPERS
 
+function buildFormHelper() {
+  preventPickingPastDates()
+  buildDropDown()
+  displayDestinationOptions(allDestinations.destinationsRepo)
+}
```

Comment:
Your error handling is next level. I should have talked to you after missing most of that lesson because you clearly were paying attention